packages/next-swc/crates/next-core/src/app_structure.rs

```diff
@@ -467,7 +474,7 @@ async fn directory_tree_to_entrypoints_internal(
                 .cell()
             } else {
                 LoaderTree {
-                    segment: directory_name.to_string(),
+                    segment: get_underscore_normalized_path(directory_name),
```

Comment:
`get_underscore_normalized_path` can also be in `get_directory_tree` by changing the `basename` directly.